Added a new configurable NO_LOG_INIT to useradd

AZaugg opened the pull request.

AZaugg commented:
Adding NO_LOG_INIT parameter to be configurable by /etc/default/useradd. 

$ src/useradd -D
GROUP=100
HOME=/home
INACTIVE=-1
EXPIRE=
SHELL=/bin/bash
SKEL=/etc/skel
CREATE_MAIL_SPOOL=no
NO_LOG_INIT=no

$ cat /etc/default/useradd
# useradd defaults file
GROUP=100
HOME=/home
INACTIVE=-1
EXPIRE=
SHELL=/bin/bash
SKEL=/etc/skel
CREATE_MAIL_SPOOL=no

$ src/useradd -D
GROUP=100
HOME=/home
INACTIVE=-1
EXPIRE=
SHELL=/bin/bash
SKEL=/etc/skel
CREATE_MAIL_SPOOL=no
NO_LOG_INIT=yes

$ cat /etc/default/useradd
# useradd defaults file
GROUP=100
HOME=/home
INACTIVE=-1
EXPIRE=
SHELL=/bin/bash
SKEL=/etc/skel
CREATE_MAIL_SPOOL=no
NO_LOG_INIT=yes
CREATE_MAIL_SPOOL=no

Adding this option will allow a person to control the default behaviour when new users are added.

vapier commented:
can we please avoid negative variable names.  they are hard to read & reason about and easy to fudge.

so name it `LOG_INIT` instead and invert the defaults.

AZaugg commented:
Hi @vapier 
Thanks for the feedback. I updated the behaviour and removed the negative logic.

Here is the config file with LOG_init set to yes, which is the default behavior:
[a@localhost src]$ cat /etc/default/useradd
GROUP=100
HOME=/home
INACTIVE=-1
EXPIRE=
SHELL=/bin/bash
SKEL=/etc/skel
CREATE_MAIL_SPOOL=no
LOG_INIT=yes

Now running useradd 
[a@localhost src]$ ./useradd -D
GROUP=100
HOME=/home
INACTIVE=-1
EXPIRE=
SHELL=/bin/bash
SKEL=/etc/skel
CREATE_MAIL_SPOOL=no
LOG_INIT=yes

Now setting LOG_INIT to no, to disable writing to the faillogs
[a@localhost src]$ cat /etc/default/useradd
GROUP=100
HOME=/home
INACTIVE=-1
EXPIRE=
SHELL=/bin/bash
SKEL=/etc/skel
CREATE_MAIL_SPOOL=no
LOG_INIT=no

Now running useradd
[a@localhost src]$ ./useradd -D
GROUP=100
HOME=/home
INACTIVE=-1
EXPIRE=
SHELL=/bin/bash
SKEL=/etc/skel
CREATE_MAIL_SPOOL=no
LOG_INIT=no

Now running useradd with the config option missing from the defaults file
[a@localhost src]$ cat /etc/default/useradd
GROUP=100
HOME=/home
INACTIVE=-1
EXPIRE=
SHELL=/bin/bash
SKEL=/etc/skel
CREATE_MAIL_SPOOL=no

Now running useradd
[a@localhost src]$ ./useradd -D
GROUP=100
HOME=/home
INACTIVE=-1
EXPIRE=
SHELL=/bin/bash
SKEL=/etc/skel
CREATE_MAIL_SPOOL=no
LOG_INIT=yes

hallyn commented:
patch 2/2 looks good ,thanks.

AZaugg commented on man/login.defs.d/MAIL_DIR.xml:
```diff
@@ -35,6 +35,8 @@
       when its corresponding user account is modified or deleted. If not
       specified, a compile-time default is used.
+      The parameter CREATE_MAIL_SPOOL in <filename>/etc/default/useradd</filename>
+      can be used to make determine if the mail spool should be created.
```
Sure

mminkus commented on src/useradd.c:
```diff
@@ -207,6 +208,7 @@ static bool home_added = false;
 #define DEXPIRE			"EXPIRE="
 #define DSKEL			"SKEL="
 #define DCREATE_MAIL_SPOOL	"CREATE_MAIL_SPOOL="
+#define DLOG_INIT	"LOG_INIT="
```
Not aligned.
